Remove unneeded options

FMarno opened the pull request:
Draft PR before going upstream. This is supposed to be similar to https://github.com/intel/intel-xpu-backend-for-triton/pull/1148

FMarno commented:
@sommerlukas @jopperm @mfrancepillois can you take a look at this before I create an upstream PR please?

FMarno commented:
@etiotto @whitneywhtsang here is the draft for upstreaming the device property changes FYI

sommerlukas commented on lib/Dialect/TritonGPU/Transforms/Utility.cpp:
```diff
@@ -818,6 +819,24 @@ bool isPureUnaryInlineAsm(Operation *op) {
+  StringAttr targetAttr =
+      cast<StringAttr>(module->getAttr(triton::AttrTargetName));
+
+  auto ref = targetAttr.strref();
+  assert(ref.starts_with("cuda:") &&
+         "expected target attr to be prefixed with \"cuda:\"");
+
+  auto capabilityStr = ref.drop_front(5); // drop the "cuda:"
+  int computeCapability = 80;             // default capability
+  assert(!capabilityStr.getAsInteger(10, computeCapability));
```
I wonder if this part here could be expressed more elegantly with a simple `llvm::Regex`, this would allow to check for the correct format and retrieve the capability from the right part of the string.

FMarno replied:
I personally find the code very straight forward and am not keen to change it to a regex unless there is functional change. I believe `getAsInteger` should check that the whole string (`capabilityStr`) is valid for conversion if that is your concern. From the `getAsInteger` docs

> If the string is invalid or if only a subset of the string is valid, this returns true to signify the error

sommerlukas replied:
I'm not feeling too strong about this, we can keep as is for upstream review if you prefer.

etiotto replied:
The assert should have an associates message

etiotto replied:
```
capabilityStr.getAsInteger(10, computeCapability) 
```
Should not be enclosed into an assert because is possible to compile out assertion and that code is necessary to initialize `computeCapability` to the value in the module.  Suggesting to place that line outside of the assert and in the assert check the returned value.

FMarno replied:
thanks, that might have been hard to debug

sommerlukas reviewed (commented):
Just one minor comment.

etiotto commented on include/triton/Dialect/TritonGPU/Transforms/Utility.h:
```diff
@@ -169,6 +169,9 @@ Value linearize(OpBuilder &b, Location loc, ArrayRef<Value> multiDim,
 bool isPureUnaryInlineAsm(Operation *op);
 
+// read the compute capability from the module attributes
+int getComputeCapability(Operation *module);
```
This should ultimately be moved into NVidia specific utilities because the concept of a compute capability applies only to NVidia GPUs.  For the time being I guess we can leave it here but I would perhaps rename the function to `getNVidiaComputeCapability`

FMarno replied:
Sure. Yes, I do see that as a possibly difficult thing to merge.

etiotto commented on lib/Dialect/TritonGPU/Transforms/Pipeliner/SoftwarePipeliner.cpp:
```diff
@@ -94,13 +94,7 @@ static bool pipelineLoop(scf::ForOp forOp, int numStages) {
-    this->numCTAs = numCTAs;
-    this->computeCapability = computeCapability;
-  }
+  PipelinePass(int numStages) { this->numStages = numStages; }
```
Is it possible to use the constructor init list to initialize `numStages` ? 
```
  PipelinePass(int numStages) : numStages(numStages) {}
```

FMarno replied:
no longer applicable after merging main

etiotto commented on lib/Dialect/TritonGPU/Transforms/Utility.cpp:
```diff
@@ -818,6 +819,24 @@ bool isPureUnaryInlineAsm(Operation *op) {
+  StringAttr targetAttr =
+      cast<StringAttr>(module->getAttr(triton::AttrTargetName));
+
+  auto ref = targetAttr.strref();
```
Suggest to use concrete type instead of auto here and at line 833

FMarno replied:
done

int3 committed:
[AUTOTUNER] Don't cache benchmarking stream (#3993)

This caching seems to be responsible for some CUDA OOMs we encountered
in Meta-internal builds. I haven't got a reduced repro, but this change
does seem to fix things. My hypothesis is that the cached stream is
causing the memory allocated for the graph to be retained.

lancerts committed:
[TUTORIALS] Update link and path in tutorial 04 (#3974)

pawelszczerbuk committed:
Hoist convert_layout above extending fp_to_fp (#3995)

We are normally hoisting layout conversions above operations that may
make the convert more expensive, like Ext and broadcasts, but we have
been missing the case of `FpToFpOp` that can be extending too.

pawelszczerbuk committed:
Allow FP8 types in `arith.constant` initializers (#3996)

There was a bug in lowering `arith.constant` where only "regular" FP
types and BF16 were allowed as initializers (we couldn't even lower 0
init for FP8).
I have included all the FP8 types in the `isFloat` helper. Let me know
if it makes sense.

pawelszczerbuk committed:
`If` pipelining fixes (#3998)

Two fixes around pipelining `if`s:
* Schedule prologue and epilogue dependencies in stage order - we should
always do that (see `scheduleDependencies`) otherwise we may end up with
incorrect schedule
* Properly handle if predication (needed if there are memory ops in the
`if`)

Jokeren committed:
[CI] Remove `--no-build-isolation` (#3999)

https://github.com/LLNL/hatchet/pull/125#issuecomment-2130102636

Previously `no-build-isolation` is used to take advantage of the
incremental build.
Now that we have containerized the CI, it probably doesn't help much.

ThomasRaoux committed:
Add missing core maintainer (#4001)

christopherhesse committed:
[DOCS] fixes to tl.dot (#4006)

The `tl.dot` docs have a couple of typos and seem to be missing mention
of lower precision dtypes

chsigg committed:
Remove odd character at beginning of file (#4011)

Co-authored-by: Tori <vwbaker@google.com>

vwbaker committed:
[BACKEND] Update LLVM version to https://github.com/llvm/llvm-project/commit/10dc3a8e916d73291269e5e2b82dd22681489aa1 (#4010)

Paran0idy committed:
[DOCS] improve Triton Linear layout doc (#4005)

- Correct some indices errors.

FMarno commented on lib/Dialect/TritonGPU/Transforms/Utility.cpp:
```diff
@@ -818,6 +819,25 @@ bool isPureUnaryInlineAsm(Operation *op) {
+  StringRef capabilityStr = ref.drop_front(5); // drop the "cuda:"
+  int computeCapability = 80;                  // default capability
+  bool parseError = capabilityStr.getAsInteger(10, computeCapability);
+  assert(!parseError && "valid compute capability string in target attribute");
```
```suggestion
  assert(!parseError && "invalid compute capability string in target attribute");
```

FMarno committed:
Remove unneeded options from pass

read the same information from the module op attributes instead

Signed-off-by: Finlay Marno <finlay.marno@codeplay.com>
